src/tribol/interface/tribol.cpp

```diff
@@ -583,83 +725,130 @@ int getCSRMatrix( int** I, int** J, real** vals, int csId,
-            mesh.m_nodalFields.m_node_gap = fieldVariable;
-            mesh.m_nodalFields.m_is_node_gap_set = true;
-         }
-         break;
-      }
-      case MORTAR_PRESSURES:
-      {
-         if (fieldVariable==nullptr)
-         {
-            SLIC_WARNING( "tribol::registerRealNodalField(): null pointer to data for " << 
-                          "'MORTAR_PRESSURES' on mesh, " << meshId << ".");
-            mesh.m_isValid = false;
-         }
-         else
-         {
-            mesh.m_nodalFields.m_node_pressure = fieldVariable;
-            mesh.m_nodalFields.m_is_node_pressure_set = true;
-         }
-         break;
-      }
-      default:
-      {
-         SLIC_ERROR( "tribol::registerRealNodalField(): the field argument " << 
-                     "for mesh, " << meshId << ", is not an accepted tribol nodal field." );
-      }
-   } // end switch over field
+      SLIC_WARNING( "tribol::registerMortarGaps(): null pointer to data " << 
+                    "on mesh " << meshId << ".");
+      mesh.m_isValid = false;
+   }
+   else
+   {
+      mesh.m_nodalFields.m_node_gap = nodal_gaps;
+      mesh.m_nodalFields.m_is_node_gap_set = true;
+   }
+   
+}
 
-} // end registerRealNodalField()
+//------------------------------------------------------------------------------
+mfem::ParGridFunction getMfemGap( integer cs_id )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemSubmeshData(), 
+      "Coupling scheme does not contain MFEM dual field data. "
+      "Was the coupling scheme created with registerParMesh() and is the "
+      "enforcement method LAGRANGE_MULTIPLIER?"
+   );
+   return coupling_scheme->getMfemSubmeshData()->GetSubmeshGap();
+}
+
+//------------------------------------------------------------------------------
+void registerMortarPressures( integer meshId,
+                              const real * nodal_pressures )
+{
+   MeshManager & meshManager = MeshManager::getInstance();
+
+   SLIC_ERROR_IF(!meshManager.hasMesh(meshId), "tribol::registerMortarPressures(): " << 
+                 "no mesh with id " << meshId << " exists.");
+
+   MeshData & mesh = meshManager.GetMeshInstance( meshId );
+
+   if (nodal_pressures == nullptr)
+   {
+      SLIC_WARNING( "tribol::registerMortarPressures(): null pointer to data " << 
+                    "on mesh " << meshId << ".");
+      mesh.m_isValid = false;
+   }
+   else
+   {
+      mesh.m_nodalFields.m_node_pressure = nodal_pressures;
+      mesh.m_nodalFields.m_is_node_pressure_set = true;
+   }
+   
+}
+
+//------------------------------------------------------------------------------
+mfem::ParGridFunction& getMfemPressure( integer cs_id )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemSubmeshData(), 
+      "Coupling scheme does not contain MFEM dual field data. "
```

Comment:
Consider clarifying dual field here. The pressure (LMs) is the dual field so this makes sense, but not everyone will understand this. Maybe just say "pressures" since that is what the host-code user thinks they are getting by calling this routine.

Reply:
Updated to pressure